swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -727,6 +747,30 @@ predicate storeStep(Node node1, ContentSet c, Node node2) {
     c.isSingleton(any(Content::ArrayContent ac))
   )
   or
+  // read of a dictionary value via subscript operator, with intermediate step
+  exists(AssignExpr assign, SubscriptExpr subscript |
+    subscript = assign.getDest() and
+    (
+      subscript.getArgument(0).getExpr() = node1.asExpr() and
+      node2.(DictionarySubscriptNode).getExpr() = subscript and
+      c.isSingleton(any(Content::TupleContent tc | tc.getIndex() = 1))
+      or
+      assign.getSource() = node1.asExpr() and
+      node2.(DictionarySubscriptNode).getExpr() = subscript and
+      c.isSingleton(any(Content::TupleContent tc | tc.getIndex() = 1))
+      or
+      node1.(DictionarySubscriptNode) = node1 and
+      node2.asExpr() = subscript and
+      c.isSingleton(any(Content::CollectionContent cc))
+    )
+  )
+  or
+  exists(DictionaryExpr dict |
+    node1.asExpr() = dict.getAnElement() and
+    node2.asExpr() = dict and
+    c.isSingleton(any(Content::CollectionContent cc))
+  )
```

Comment:
What's the difference between `SubscriptExpr` and `DictionaryExpr`?  Is this for initializing a dictionary?  (perhaps it should be documented somewhere)

Reply:
`SubscriptExpr` is `dict[key]`, `DicitonaryExpr` is `[key:value, ...]`

Reply:
Thanks.